server/apps/pa_img/pa_img_datalayer.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014, 2015 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import datetime
+from io import BytesIO
+import json
+import logging
+import re
+
+from eve.io.base import DataLayer
+from eve_elastic.elastic import ElasticCursor
+from flask import url_for
+import urllib3
+
+from superdesk.errors import SuperdeskApiError
+from superdesk.media.media_operations import process_file_from_stream, decode_metadata
+from superdesk.media.renditions import generate_renditions, delete_file_on_error
+from superdesk.metadata.item import ITEM_TYPE, CONTENT_TYPE
+from superdesk.utc import utc, utcnow
+
+
+urllib3.disable_warnings()
+
+logger = logging.getLogger(__name__)
+
+
+def extract_params(query, names):
+    findall = re.findall('([\w]+):\(([\w\s]+)\)', query)
+    params = {name: value for (name, value) in findall if name in names}
+    result = re.search('^\(([\w\s]+)\)', query)
+    if result:
+        params['text'] = result.group(1)
+    else:
+        result = re.search('[^:]+\(([\w\s]+)\)', query)
+        if result:
+            params['text'] = result.group(1)
+    return params
+
+
+class PaImgDatalayer(DataLayer):
```

Comment:
The part that is common to 'search ingest' providers is on SearchIngestService that is used now for both aap_mm and pa_img. I think for now to extend the DataLayer to implement the custom access to media storage service should be ok. It can be refactored easy later if it will be required.